client/modules/datafiles/src/DatafilesToolbar/DatafilesToolbar.tsx

```diff
@@ -73,6 +130,14 @@ export const DatafilesToolbar: React.FC = () => {
             </ToolbarButton>
           )}
         </DatafilesModal.Copy>
+        <ToolbarButton
```

Comment:
Let's extract the trash button into its own component, since it has as much business logic as the rest of the toolbar code combined.